### ISSUE TITLE
if (!strcmp()) flagged as "inverted logic" when there's an else block

### ISSUE THREAD

**Issue by Max-E** [oclint/oclint#55]
This occurs in OClint version 0.6:

```
maxtothemax@maxtothemax-mint ~ $ oclint -version
LLVM (http://llvm.org/):
  LLVM version 3.2svn
  Optimized build.
  Built Dec 28 2012 (23:04:17).
  Default target: i386-pc-linux-gnu
  Host CPU: corei7
```

Here's a short but complete program that triggers the bug:

``` c
#include <stdio.h>
#include <string.h>

int main (int argc, char *argv[]) 
{
    if (!strcmp ("string", "string"))
        printf ("string equals string\n");
    else
        printf ("can't happen\n");
    return 0;
}
```

It produces the following output:

```
$ oclint oclint_strcmp_inverting.c --
Processing: /home/maxtothemax/code/wtf/oclint_strcmp_inverting.c.
OCLint Report

Summary: TotalFiles=1 FilesWithViolations=1 P1=0 P2=2 P3=1 

/home/maxtothemax/code/wtf/oclint_strcmp_inverting.c:6:9: inverted logic P3 
/home/maxtothemax/code/wtf/oclint_strcmp_inverting.c:4:11: unused method parameter P2 
/home/maxtothemax/code/wtf/oclint_strcmp_inverting.c:4:21: unused method parameter P2 

[OCLint (http://oclint.org) v0.6]
```

OCLint correctly doesn't complain if there's no "else" block. However, I'd argue that even if there is one, !strcmp() is idiomatic in C and shouldn't be considered a style issue.
